chore: add dependabot configuration for automated dependency updates

netbonus opened the pull request:
- Daily security patches with grouped PRs
- Weekly individual dependency updates
- Monthly GitHub Actions updates
- Individual PRs for granular review and testing

yilmazbahadir commented on .github/dependabot.yml:
```diff
@@ -0,0 +1,34 @@
+    directory: "/"
+    schedule:
+      interval: "weekly"
+    open-pull-requests-limit: 10
```
> increase-if-necessary: Only update if required to satisfy constraints
Only bumps versions when needed by your dependency constraints. For example, if your package.json has "lodash": "^4.17.0" and both 4.17.21 and 5.0.0 are available, it will only update to 4.17.21 (not the major    version 5.0.0) since your semver range is satisfied.
  
```suggestion
    versioning-strategy: "increase-if-necessary"
    open-pull-requests-limit: 10
```

yilmazbahadir commented on .github/dependabot.yml:
```diff
@@ -0,0 +1,34 @@
+    schedule:
+      interval: "daily"
+    open-pull-requests-limit: 5
+    labels:
```
```suggestion
    commit-message:
        prefix: "deps"
        prefix-development: "deps"
        include: "scope"
    labels:
```

yilmazbahadir commented on .github/dependabot.yml:
```diff
@@ -0,0 +1,34 @@
+    schedule:
+      interval: "weekly"
+    open-pull-requests-limit: 10
+    labels:
```
```suggestion
    commit-message:
        prefix: "deps"
        prefix-development: "deps"
        include: "scope"
    labels:
```

yilmazbahadir commented on .github/dependabot.yml:
```diff
@@ -0,0 +1,34 @@
+    schedule:
+      interval: "monthly"
+    open-pull-requests-limit: 3
+    labels:
```
```suggestion
    commit-message:
        prefix: "ci"
        include: "scope"
    labels:
```

netbonus committed:
chore: configure dependabot for automated dependency updates

- Daily npm dependency checks with grouped security updates
- Monthly GitHub Actions updates
- Individual PRs for non-security npm updates
